src/Components/Analyzers/test/TestFiles/ComponentInternalUsageDiagnosticsAnalyzerTest/UsesRendererTypesInDeclarations.cs

```diff
@@ -22,11 +22,15 @@ protected override void HandleException(Exception exception)
             throw new NotImplementedException();
         }
 
-        /*MMParameter*/protected override Task UpdateDisplayAsync(in RenderBatch renderBatch) 
+        /*MMParameter*/protected override Task UpdateDisplayAsync(in RenderBatch renderBatch)
         {
             throw new NotImplementedException();
         }
 
         /*MMReturnType*/private Renderer GetRenderer() => _field;
+
+        public interface ITestInterface
```

Comment:
FYI @rynowak \ @SteveSandersonMS \ @mkArtakMSFT : 

`type.BaseType` for interface types is null: https://github.com/dotnet/aspnetcore/pull/18608/files#diff-f7300f2c3fe64cdbb8f5dfd7854b0b56R85. This causes the call to `VisitDeclarationSymbol` to null-ref. It looks like the compiler was swallowing this in the past, but they now are presented as warnings. We were able to author the unit test to trivially cause this to fail. Should we consider patching this?